### ISSUE TITLE
any_spawner does not execute local futures when using futures-executor

### ISSUE THREAD

**Issue by stefnotch** [leptos-rs/leptos#3083]
**Describe the bug**

The following code will never print "hello from local". As far as I can tell, the [`LocalPool`](https://docs.rs/futures/latest/futures/executor/struct.LocalPool.html) needs to be polled to make any progress.
```rust
Executor::init_futures_executor().expect("couldn't set executor");
Executor::spawn_local(async {
    println!("hello from local");
});
```

**Leptos Dependencies**

`any_spawner = { version = "0.1.1", features = ["futures-executor"] }`

**To Reproduce**
```
#[cfg(feature = "futures-executor")]
#[test]
fn can_make_local_progress() {
    use crate::Executor;
    use std::sync::{atomic::AtomicUsize, Arc};
    Executor::init_futures_executor().expect("couldn't set executor");
    let counter = Arc::new(AtomicUsize::new(0));
    Executor::spawn_local({
        let counter = Arc::clone(&counter);
        async move {
            assert_eq!(
                counter.fetch_add(1, std::sync::atomic::Ordering::AcqRel),
                0
            );
        }
    });

    // TODO: Convince the local executor to make progress. Currently not possible, as far as I can tell.

    assert_eq!(counter.load(std::sync::atomic::Ordering::Acquire), 1);
}
```

**Expected behavior**
I expected there to be a way of driving progress forward when using `futures-executor`

**Comment by gbj** [comment]
Feel free to make a PR. I added the `futures-executor` support basically playing around, and have no use for it.

**Comment by stefnotch** [comment]
I tried doing so, but my conclusion is that it's not really possible. Essentially the `futures-executor` API comes with the constraint that `spawn_local(async { spawn_local(async { })})` does not work. 

Here's a commit where I attempted a dirty implementation and noticed the issue.
https://github.com/leptos-rs/leptos/commit/8f26ce002e9f983a2245708563637a0ffe9350e7

So I think the correct move is to remove the `futures-executor`
